### DISCUSSION TITLE
Good way to show 'saving' between your tariff and override

### DISCUSSION THREAD

**Discussion by JRascagneres** [BottlecapDave/HomeAssistant-OctopusEnergy#387] [Q&A]
I made the following sensor:
```
    - name: "Agile Cost Saving"
      unit_of_measurement: 'GBP'
      device_class: "monetary"
      state_class: "total"
      unique_id: "agile_cost_saving"
      state: "{{ states('sensor.octopus_energy_electricity_[redacted]_previous_accumulative_cost_override') | float - states('sensor.electricity_[redacted]_previous_accumulative_cost') | float }}"
```

However due to the slightly differing update times I get the following:
![image](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/3264971/778ddec7-fb39-47d0-b4d7-2c79b286dab9)

Anyone got any good ideas how I can stop the random little spike that occurs?

**Comment by BottlecapDave** [comment]
Hello.

You could potentially compare the `last_calculated_timestamp` on both of the sensors and only show the difference if both match. However the question is what to do when they don't match.

The best way of solving your issue that I can think of would be having a custom [input number](https://www.home-assistant.io/integrations/input_number/), which you set to your state value via an automation when either state changes with a condition that the `last_calculated_timestamp` attributes match. This way it will keep the old value until it's time to recalculate the saving.

**Reply by JRascagneres** [reply]
Thanks. Will give it a play. Don't suppose this could be something that could be implemented into the integration - Specifically a diff between the override and actual costs?
